src/buffered-process.coffee

```diff
@@ -159,6 +160,22 @@ class BufferedProcess
           process.kill(pid)
       @killProcess()
 
+  # Kill all child processes spawned by a spawned child process
+  #
+  # Uses ps-tree, so it works on *nix systems only
+  killOnNix: ->
+    return unless @process?
+
+    parentPid = @process.pid
+    psTree parentPid, (err, children) ->
+        pidsToKill = children.map (p) -> p.PID
+        for pid in pidsToKill
+          try
+            process.kill(pid)
+        return
+
+    @killProcess()
```

Comment:
Given the fact the `psTree` call is async, I think this invocation of `@killProcess` should be nested inside the callback handler (as seen in https://github.com/atom/atom/pull/9108#issuecomment-147311776). This is to guard against any possible side-effects of killing the primary process first. What do you think?